src/controllers/legacycontrollersettings.h

```diff
@@ -392,12 +500,45 @@ class LegacyControllerEnumSetting
         return QJSValue(stringify());
     }
 
-    const QList<Item>& options() const {
+#ifdef MIXXX_USE_QML
+    void setValue(const QJSValue& value) override {
```

Comment:
A check is missing if QJSValue has a string. It may already has the value (number)?

Reply:
No, the value is always the value that needs to be find in the list of valid option, and this list of option is always a string. For the same reason than colour, we attempt to infer the value into string even if it isn't, for convenience. 
See the `Q_PROPERTY` definition in the mixin.

If one want to set the index directly, they would use the `currentIndex` property (and thus the `setCurrentIndex`)

Reply:
No, it's not - "Inside value = int" is an implementation detail that caller don't have to care about. The caller will always work with "value = string" so this is as intended.